hydra-cluster/bench/Bench/EndToEnd.hs

```diff
@@ -238,20 +298,22 @@ movingAverage confirmations =
 -- transaction is returned.
 seedNetwork :: RunningNode -> Dataset -> Tracer IO FaucetLog -> IO TxId
 seedNetwork node@RunningNode{nodeSocket, networkId} Dataset{fundingTransaction, clientDatasets} tracer = do
-  fundClients
-  forM_ clientDatasets fuelWith100Ada
+  fundClients networkId nodeSocket fundingTransaction
+  forM_ (clientKeys <$> clientDatasets) (fuelWith100Ada tracer node)
   putTextLn "Publishing hydra scripts"
   publishHydraScriptsAs node Faucet
- where
-  fundClients = do
-    putTextLn "Fund scenario from faucet"
-    submitTransaction networkId nodeSocket fundingTransaction
-    void $ awaitTransaction networkId nodeSocket fundingTransaction
-
-  fuelWith100Ada ClientDataset{clientKeys = ClientKeys{signingKey}} = do
-    let vk = getVerificationKey signingKey
-    putTextLn $ "Seed client " <> show vk
-    seedFromFaucet node vk 100_000_000 tracer
+
+fundClients :: NetworkId -> SocketPath -> Tx -> IO ()
+fundClients networkId nodeSocket fundingTransaction = do
+  putTextLn "Fund scenario from faucet"
+  submitTransaction networkId nodeSocket fundingTransaction
+  void $ awaitTransaction networkId nodeSocket fundingTransaction
```

Comment:
Having this in a separate function is pointless. The `putTextLn` will rarely fit and it's basically an alias for `submitTransaction >=> void . awaitTransaction`

Should: keep `fuelWith100Ada` local to `seedNetwork` and instead just submit and await where you called `fundClients`